quic/s2n-quic-core/src/recovery/cubic.rs

```diff
@@ -459,16 +452,20 @@ impl CongestionController for CubicCongestionController {
     //# A PL that maintains the congestion window in terms of a limit to
     //# the number of outstanding fixed-size packets SHOULD adapt this
     //# limit to compensate for the size of the actual packets.
+
+    //= https://www.rfc-editor.org/rfc/rfc9002#section-7.2
+    //= type=exception
+    //= reason=The maximum datagram size remains at the minimum (1200 bytes) during the handshake
+    //# If the maximum datagram size is decreased in order to complete the
+    //# handshake, the congestion window SHOULD be set to the new initial
+    //# congestion window.
     #[inline]
     fn on_mtu_update(&mut self, max_datagram_size: u16) {
         let old_max_datagram_size = self.max_datagram_size;
         self.max_datagram_size = max_datagram_size;
         self.cubic.max_datagram_size = max_datagram_size;
 
-        if max_datagram_size < old_max_datagram_size {
-            self.congestion_window =
-                CubicCongestionController::initial_window(max_datagram_size) as f32;
-        } else {
+        if max_datagram_size > old_max_datagram_size {
```

Comment:
I don't think we want this check, since we try to keep the number of packets in the congestion window the same, regardless of the MTU.

Reply:
We still need the correct (reduced) MTU so that the congestion window is calculated correctly

Reply:
OK. I updated the code.